lib/mongo/database/view.rb

```diff
@@ -55,7 +62,8 @@ def collection_names(options = {})
         @batch_size = options[:batch_size]
         session = client.send(:get_session, options)
         cursor = read_with_retry_cursor(session, ServerSelector.primary, self) do |server|
-          send_initial_query(server, session, name_only: true)
+          options[:name_only] = true
```

Comment:
This mutates options. We generally do not mutate arguments to methods. A possible way of spelling this is `options = options.merge(name_only: true)`.

Reply:
What is the difference functionality-wise between my method and yours?